src/Daemon/Daemon.cpp

```diff
@@ -252,9 +258,10 @@ namespace usbguard
   void Daemon::loadRules(const String& path)
   {
     USBGUARD_LOG(Info) << "Loading permanent policy file " << path;
-    std::shared_ptr<RuleSet> ptr(new RuleSet(this));
-    _policy.setRuleSet(ptr);
-    _policy.load(path);
+
+    auto ruleset = _nss.getRuleSet(this);
+    _policy.setRuleSet(ruleset);
+    ruleset->load(path);
```

Comment:
From a design perpective I would load the ruleset first, then call setRuleSet, as the load could throw an exception.